cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/prometheus/PrometheusMetricSamplerTest.java

```diff
@@ -0,0 +1,491 @@
+                null,
+                null
+            ), Arrays.asList(new PrometheusValue(START_EPOCH_SECONDS, FIXED_VALUE))));
+        }
+        return resultList;
+    }
+
+    private List<PrometheusQueryResult> buildTopicResults() {
+        List<PrometheusQueryResult> resultList = new ArrayList<>();
+        for (int brokerId = 0; brokerId < TOTAL_BROKERS; brokerId++) {
+            resultList.add(new PrometheusQueryResult(new PrometheusMetric(
+                "broker-" + brokerId + ".test-cluster.org:11001",
+                TEST_TOPIC,
+                null
+            ), Arrays.asList(new PrometheusValue(START_EPOCH_SECONDS, FIXED_VALUE))));
+        }
+        return resultList;
+    }
+
+    private List<PrometheusQueryResult> buildTopicResultsWithNullTopic() {
+        List<PrometheusQueryResult> resultList = new ArrayList<>();
+        for (int brokerId = 0; brokerId < TOTAL_BROKERS; brokerId++) {
+            resultList.add(new PrometheusQueryResult(new PrometheusMetric(
+                "broker-" + brokerId + ".test-cluster.org:11001",
+                null,
+                null
+            ), Arrays.asList(new PrometheusValue(START_EPOCH_SECONDS, FIXED_VALUE))));
+        }
+        return resultList;
+    }
+
+    private List<PrometheusQueryResult> buildPartitionResultsWithNullPartition() {
+        List<PrometheusQueryResult> resultList = new ArrayList<>();
+        for (int brokerId = 0; brokerId < TOTAL_BROKERS; brokerId++) {
+            resultList.add(new PrometheusQueryResult(new PrometheusMetric(
+                "broker-" + brokerId + ".test-cluster.org:11001",
+                TEST_TOPIC,
+                null
+            ), Arrays.asList(new PrometheusValue(START_EPOCH_SECONDS, FIXED_VALUE))));
+        }
+        return resultList;
+    }
+
+    private List<PrometheusQueryResult> buildPartitionResultsWithMalformedPartition() {
+        List<PrometheusQueryResult> resultList = new ArrayList<>();
+        for (int brokerId = 0; brokerId < TOTAL_BROKERS; brokerId++) {
+            resultList.add(new PrometheusQueryResult(new PrometheusMetric(
+                "broker-" + brokerId + ".test-cluster.org:11001",
+                TEST_TOPIC,
+                "non-number"
+            ), Arrays.asList(new PrometheusValue(START_EPOCH_SECONDS, FIXED_VALUE))));
+        }
+        return resultList;
+    }
+
+    private List<PrometheusQueryResult> buildPartitionResults() {
+        List<PrometheusQueryResult> resultList = new ArrayList<>();
+        for (int brokerId = 0; brokerId < TOTAL_BROKERS; brokerId++) {
+            for (int partition = 0; partition < TOTAL_PARTITIONS; partition++) {
+                resultList.add(new PrometheusQueryResult(new PrometheusMetric(
+                    "broker-" + brokerId + ".test-cluster.org:11001",
+                    TEST_TOPIC,
+                    String.valueOf(partition)
+                ), Arrays.asList(new PrometheusValue(START_EPOCH_SECONDS, FIXED_VALUE))));
```

Comment:
Nit: Applies to similar uses above: Should this be `Collections.singletonList` since it has a single element?

Reply:
I have deliberately kept it as `Arrays.asList` despite some IntelliJ warnings, because there is nothing explicit about the test that requires it to be a single value. I am, however, do not have a strong opinion on this, and can change it to `Collections.emptyList()`.

Reply:
Changed to `Collections.emptyList()` everywhere.